indigo_api/models/works.py

```diff
@@ -312,6 +313,27 @@ def friendly_type(self):
         if plugin:
             return plugin.work_friendly_type(self)
 
+    def possible_expression_dates(self):
+        initial = self.publication_date or self.commencement_date
+        amendment_dates = [a.date for a in self.amendments.all()]
+        consolidation_dates = [c.date for c in self.arbitrary_expression_dates.all()]
+        dates = [
+            {'date': date,
+             'amendment': date in amendment_dates,
+             'consolidation': date in consolidation_dates,
+             'initial': date == initial}
+            for date in amendment_dates + consolidation_dates
```

Comment:
There could be duplicate dates. I suggest doing something like `all_dates = set(amendment_dates + consolidation_dates)`